enhancements/testing/ephemeral-agent-installer-e2e-tests.md

```diff
@@ -0,0 +1,337 @@
+
+### User Stories
+
+#### Run Assisted Installer Outside k8s
+
+As a developer, I need to run Assisted Installer without installing a k8s platform first 
+so that I can test the installation of a users first cluster, cluster 0.
+
+This is the overall purpose of this ephemeral agent based installer image project. 
+Unfortunately, neither of the existing frameworks actually can actually assist with this. 
+
+* The Assisted Installer e2e tests that run in dev-scripts first use the IPI installer 
+to install OCP and then the infrastructure-operator on this OCP cluster making it a hub 
+cluster.
+
+* The Assisted Installer e2e tests that run in assisted-test-infra installs minikube first 
+and then runs Assisted Installer on top of it.
+
+Which means in both existing e2e frameworks we first setup some form of k8s first. Given
+the goal of the ephemeral agent based installer project to install a users first cluster, 
+cluster 0, this is highly undesirable.
+
+While there is a flow to run Assisted Installer in podman this is a manually supported 
+flow. The support of it in the e2e frameworks is mostly abandoned by the Assisted Installer 
+team in favor of infrastructure-operator and the hosted SaaS solution.
+
+#### Setup Environment for Install Without Installing OCP or k8s
+
+As a developer, I need to set up VMs so that I can simulate the installation of a users 
+first cluster using the ephemeral based agent installer image.
+
+Both existing frameworks today set up VMs using libvirt today to test a users OCP install.
+However, they also both immediately use those VMs to start the installation process. There
+isn't an easy way to stop the current frameworks from using IPI and Assisted Installer
+after the VM creation.
+
+
+#### Launch a VM with the Ephemeral Agent Based Installer Image
+
+As a developer, I need to launch a VM with the built ephemeral agent based installer
+image so that I can test and validate this image.
+
+Both existing frameworks don't have an easy way to do this. Which makes sense because
+they predate the ephemeral agent based installer project. There also isn't an easy way
+to use an existing routine to just launch a generic ISO image which would have been
+something we could have re-purposed. 
+
+### API Extensions
+
+Not applicable at this time.
+
+### Implementation Details/Notes/Constraints [optional]
+
+### Risks and Mitigations
+
+The direction of this proposal is now to extend and refactor parts of dev-scripts
+for the agent-based installer.
+
+** There will be now a third team using dev-scripts for slightly different purposes.
+Which means the risk of breaking things in the central framework now is larger and
+has a larger "splash zone."
+
+We can mitigate this for the new ephemeral agent installer by adding tests into the
+dev-scripts repository itself. The purpose of this would be to find breakages in the
+tests across the different teams.
+
```

Comment:
Another risk is that dev-scripts relies on an upstream repo, metal3-dev-env, that isn't completely within our control and changes there can break downstream jobs. I don't think that has happened very often, but see https://coreos.slack.com/archives/C01RJHA6BRC/p1646755848747089 for one example.

Reply:
Doesn't happen often but has happened in the past year maybe 2-3 times from the Assisted Installer front, I believe.

Reply:
It doesn't seem a very common case, also per my experience. As previously mentioned, the metal3-dev-env pin and devscripts CI jobs help in mitigating such risk